pyrokinetics/gene.py

```diff
@@ -0,0 +1,368 @@
+            gene_input['box']['nkx'] = numerics['nkx']
+
+        else:
+            gene_input['general']['nonlinear'] = False
+
+        # Currently forces NL sims to have nperiod = 1
+        gene_nml = f90nml.Namelist(gene_input)
+        gene_nml.float_format = pyro.float_format
+        gene_nml.write(path_to_file, force=True)
+
+    
+    def load_local_geometry(self, pyro, gene):
+        """
+        Loads local geometry 
+        """
+
+        if pyro.local_geometry_type == 'Miller':
+            self.load_miller(pyro,gene)
+
+    def load_miller(self, pyro, gene):
+        """
+        Load Miller object from GENE file
+        """
+
+        # Set some defaults here
+        gene['geometry']['magn_geometry'] = 'miller'
+
+        pyro_gene_miller = self.pyro_to_gene_miller()
+
+        miller = pyro.local_geometry
+
+        for key, val in pyro_gene_miller.items():
+            miller[key] = gene[val[0]][val[1]]
+
+        miller.kappri = miller.s_kappa*miller.kappa / miller.rho
+        miller.tri = np.arcsin(miller.delta)
+
+        # Get beta normalised to R_major(in case R_geo != R_major)
+        beta = gene['general']['beta'] * (miller.Rmaj/miller.Rgeo)**2
+
+        # Assume pref*8pi*1e-7 = 1.0
+        if beta != 0.0:
+            miller.B0 = np.sqrt(1.0/beta**0.5)
+            # Can only know Bunit/B0 from local Miller
+            miller.Bunit = miller.get_bunit_over_b0() * miller.B0
+
+        else:
+            # If beta = 0
+            miller.B0 = None
+            miller.Bunit = None
+
+        pyro.miller = miller
+
+    def load_local_species(self, pyro, gene):
+        """
+        Load LocalSpecies object from GENE file
+        """
+        nspec = gene['box']['n_spec']
+        pyro_gene_species = self.pyro_to_gene_species()
+
+        # Dictionary of local species parameters
+        local_species = LocalSpecies()
+        local_species['nspec'] = nspec
+        local_species['nref'] = None
+        local_species['names'] = []
+
+        ion_count = 0
+
+        pressure = 0.0
+        a_lp = 0.0
+        # Load each species into a dictionary
+        for i_sp in range(nspec):
+
+            species_data = CleverDict()
+
+            gene_key = 'species'
+
+            gene_data = gene[gene_key][i_sp]
+ 
+            gene_type = gene_data['name']
+
+            for pyro_key, gene_key in pyro_gene_species.items():
+                species_data[pyro_key] = gene_data[gene_key]
+
+            species_data['vel'] = 0.0
+            species_data['a_lv'] = 0.0
+            species_data['nu'] = 0.0
+            
+            if species_data.z == -1:
+                name = 'electron'
+                te = species_data.temp
+                ne = species_data.dens
+            else:
+                ion_count += 1
+                name = f'ion{ion_count}'
+
+            pressure += species_data.temp * species_data.dens
+            a_lp += species_data.temp * species_data.dens * (species_data.a_lt + species_data.a_ln)
```

Comment:
@dake0795 Look at commit ab1362b

Need to force the normalisation to electron temp/dens